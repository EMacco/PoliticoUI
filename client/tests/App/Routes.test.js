import React from 'react';
import { BrowserRouter, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Routes, { refreshPage } from '@components/App';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe('Routes', () => {
  let store, history;

  beforeEach(() => {
    store = mockStore({
      auth: {},
      dashboard: {},
      parties: {},
      offices: {}
    });
    history = {
      push: jest.fn()
    };
  });

  it('should render index page', () => {
    const comp = (
      <Provider store={store}>
        <BrowserRouter>
          <Switch>
            <Routes dispatch={jest.fn()} />
          </Switch>
        </BrowserRouter>
      </Provider>
    );

    const wrapper = mount(comp);

    expect(wrapper.find('Home')).toBeTruthy();
  });

  it('should refresh page', () => {
    localStorage.jwtToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjp7ImlkIjoxLCJmaXJzdG5hbWUiOiJFbW1hbnVlbCIsImxhc3RuYW1lIjoiT2t3YXJhIiwib3RoZXJuYW1lIjoiTmR1a2EiLCJlbWFpbCI6ImVtbWE0cmVhbDM3QGdtYWlsLmNvbSIsInBhc3N3b3JkIjoiJDJiJDEwJGdBZUFla3RWdE9xMWJkbHIuQ1hISnVHb2xTVjVTbDIvLms2VjY3NS9Qd1h0dWJjUy5QaC9tIiwicGhvbmVudW1iZXIiOiIwODEyNDE4NTMyMCIsInBhc3Nwb3J0dXJsIjoiaHR0cHM6Ly9yZXMuY2xvdWRpbmFyeS5jb20vZG40cG9rb3YwL2ltYWdlL3VwbG9hZC92MTU2MjI0OTQ0My9wdGU5Z3k0dnlxdWZhYmRzY2pmMi5qcGciLCJpc2FkbWluIjp0cnVlLCJwYXJ0eWlkIjoyfSwiaWF0IjoxNTY0MzYwNTE0LCJleHAiOjE1NjQzNjQxMTR9.PGzgzP7XC3mFLy98uchP3frG77Avh0EH5XiD_R-MrTc';
    refreshPage()
  });

  it('should render the Routes component correctly', () => {
    const shallowWrapper = shallow(<Routes />);

    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });
});
