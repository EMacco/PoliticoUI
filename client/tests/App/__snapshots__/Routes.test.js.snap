// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should render the Routes component correctly 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={true}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    role="alert"
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <BrowserRouter>
    <Switch>
      <Route
        exact={true}
        path="/"
      >
        <Connect(Landing) />
      </Route>
      <Route
        exact={true}
        path="/login"
      >
        <Connect(withRouter(Login)) />
      </Route>
      <Route
        exact={true}
        path="/register"
      >
        <Connect(withRouter(Register)) />
      </Route>
      <Route
        exact={true}
        path="/dashboard"
      >
        <Connect(UserHome) />
      </Route>
      <Route
        exact={true}
        path="/parties"
      >
        <Connect(Parties) />
      </Route>
      <Route
        exact={true}
        path="/offices"
      >
        <Connect(Offices) />
      </Route>
    </Switch>
  </BrowserRouter>
</Provider>
`;
